### ISSUE TITLE
heatmap问题

### ISSUE THREAD

**Issue by pangkun248** [lz-pku-1997/easy-pytorch-CenterNet#1]
RT,我使用了你的代码之后,发现heatmap最大值都没有超过0.1.
始终在0.7~0.8之间徘徊.我没怎么看过官方的源码。所以不太清楚问题出在哪里了。
你训练的时候没问题吗？
训练配置基本照着你给的来的。改动几乎没有

**Comment by lz-pku-1997** [comment]
> RT,我使用了你的代码之后,发现heatmap最大值都没有超过0.1.
> 始终在0.7~0.8之间徘徊.我没怎么看过官方的源码。所以不太清楚问题出在哪里了。
> 你训练的时候没问题吗？
> 训练配置基本照着你给的来的。改动几乎没有

没超过1 还是没超过0.1？

**Comment by lz-pku-1997** [comment]
> RT,我使用了你的代码之后,发现heatmap最大值都没有超过0.1.
> 始终在0.7~0.8之间徘徊.我没怎么看过官方的源码。所以不太清楚问题出在哪里了。
> 你训练的时候没问题吗？
> 训练配置基本照着你给的来的。改动几乎没有

我用了下挺正常得啊，可以在 msra_resnet.py里增加        print(np.max(self.hm(x).detach().numpy())) 查看

**Comment by lz-pku-1997** [comment]
> RT,我使用了你的代码之后,发现heatmap最大值都没有超过0.1.
> 始终在0.7~0.8之间徘徊.我没怎么看过官方的源码。所以不太清楚问题出在哪里了。
> 你训练的时候没问题吗？
> 训练配置基本照着你给的来的。改动几乎没有

刚试了下，训练过程中官方最开始也一直全是负数，训练的差不多了heatmap目标块区域才会转正，我随便打开了一张图测试了下，中心点有1.多

**Comment by pangkun248** [comment]
> > RT,我使用了你的代码之后,发现heatmap最大值都没有超过0.1.
> > 始终在0.7~0.8之间徘徊.我没怎么看过官方的源码。所以不太清楚问题出在哪里了。
> > 你训练的时候没问题吗？
> > 训练配置基本照着你给的来的。改动几乎没有
> 
> 刚试了下，训练过程中官方最开始也一直全是负数，训练的差不多了heatmap目标块区域才会转正，我随便打开了一张图测试了下，中心点有1.多

以下是我拿你给的json文件训练的结果

开始训练咯...

现在是epoch: 1 

tensor(0.1710, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:1][0/10] |loss 9.0084 |hm_loss 7.3072 |wh_loss 13.0954 |off_loss 0.3917 
tensor(0.0858, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:1][1/10] |loss 8.0220 |hm_loss 6.2755 |wh_loss 13.2291 |off_loss 0.4236 
tensor(0.0689, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:1][2/10] |loss 7.6359 |hm_loss 5.8215 |wh_loss 14.1397 |off_loss 0.4004 
tensor(0.0632, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:1][3/10] |loss 7.2239 |hm_loss 5.4650 |wh_loss 13.7410 |off_loss 0.3848 
tensor(0.0689, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:1][4/10] |loss 6.7750 |hm_loss 5.0652 |wh_loss 13.4788 |off_loss 0.3619 
tensor(0.0767, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:1][5/10] |loss 6.4148 |hm_loss 4.7415 |wh_loss 13.1247 |off_loss 0.3609 
tensor(0.0795, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:1][6/10] |loss 6.2093 |hm_loss 4.5389 |wh_loss 12.9812 |off_loss 0.3722 
tensor(0.0821, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:1][7/10] |loss 6.0360 |hm_loss 4.3717 |wh_loss 13.0306 |off_loss 0.3612 
tensor(0.0813, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:1][8/10] |loss 5.8955 |hm_loss 4.2463 |wh_loss 12.9643 |off_loss 0.3528 
tensor(0.0789, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:1][9/10] |loss 5.8497 |hm_loss 4.2012 |wh_loss 13.0051 |off_loss 0.3479 
保存成功，去看看

现在是epoch: 2 

tensor(0.0780, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:2][0/10] |loss 4.6529 |hm_loss 3.1508 |wh_loss 12.5239 |off_loss 0.2497 
tensor(0.0779, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:2][1/10] |loss 4.6758 |hm_loss 3.0932 |wh_loss 13.3175 |off_loss 0.2509 
tensor(0.0760, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:2][2/10] |loss 5.2762 |hm_loss 3.3678 |wh_loss 12.5573 |off_loss 0.6526 
tensor(0.0773, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:2][3/10] |loss 5.0610 |hm_loss 3.2824 |wh_loss 12.4964 |off_loss 0.5290 
tensor(0.0754, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:2][4/10] |loss 5.0127 |hm_loss 3.2681 |wh_loss 12.6363 |off_loss 0.4809 
tensor(0.0733, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:2][5/10] |loss 4.9422 |hm_loss 3.2305 |wh_loss 12.6305 |off_loss 0.4487 
tensor(0.0755, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:2][6/10] |loss 4.8960 |hm_loss 3.2216 |wh_loss 12.2583 |off_loss 0.4485 
tensor(0.0738, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:2][7/10] |loss 4.8124 |hm_loss 3.1870 |wh_loss 11.9701 |off_loss 0.4284 
tensor(0.0770, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:2][8/10] |loss 4.7825 |hm_loss 3.1765 |wh_loss 11.9336 |off_loss 0.4127 
tensor(0.0760, device='cuda:0', grad_fn=<MaxBackward1>)
train: [epoch:2][9/10] |loss 4.7621 |hm_loss 3.1670 |wh_loss 11.8714 |off_loss 0.4080 
保存成功，去看看

**Comment by pangkun248** [comment]
hm上的值不升反降的样子

**Comment by pangkun248** [comment]
> hm上的值不升反降的样子

忘了说我这个hm值是经过sigmoid后的

**Comment by pangkun248** [comment]
> > RT,我使用了你的代码之后,发现heatmap最大值都没有超过0.1.
> > 始终在0.7~0.8之间徘徊.我没怎么看过官方的源码。所以不太清楚问题出在哪里了。
> > 你训练的时候没问题吗？
> > 训练配置基本照着你给的来的。改动几乎没有
> 
> 刚试了下，训练过程中官方最开始也一直全是负数，训练的差不多了heatmap目标块区域才会转正，我随便打开了一张图测试了下，中心点有1.多

是我疏忽了,后处理的时候没有把坐标*4。。。现在没问题了

**Comment by lz-pku-1997** [comment]
> > > RT,我使用了你的代码之后,发现heatmap最大值都没有超过0.1.
> > > 始终在0.7~0.8之间徘徊.我没怎么看过官方的源码。所以不太清楚问题出在哪里了。
> > > 你训练的时候没问题吗？
> > > 训练配置基本照着你给的来的。改动几乎没有
> > 
> > 
> > 刚试了下，训练过程中官方最开始也一直全是负数，训练的差不多了heatmap目标块区域才会转正，我随便打开了一张图测试了下，中心点有1.多
> 
> 是我疏忽了,后处理的时候没有把坐标*4。。。现在没问题了

嗯嗯，那就好，抱歉，昨天实在没空试下和回复你hhh
